libs/blocks/global-navigation/global-navigation.js

```diff
@@ -505,8 +505,46 @@ class Gnav {
     };
 
     const onAnalyticsEvent = (data) => {
-      // TODO: update based on requirements
-      console.log('analytics data', data);
+      if (!data) return;
+
+      const getInteraction = () => {
+        const { event: { type, subtype } = {}, source: { name } = {} } = data;
+        const contentName = data.content?.name;
```

Comment:
Could you not destructure `contentName` as well? Something like `{ event: { type, subtype } = {}, source: { name } = {}, content: { name: contentName } = {} } = data;`

Reply:
Makes sense. Made the change 👍